src/components/Contact.tsx

```diff
@@ -46,10 +48,15 @@ export default function Contact() {
       message: message
     }
 
-    axios.post("https://email-free-api.herokuapp.com/send-email", formToBeSubmitted)
-    .then(successToastEmitter).catch(error => {
+    setDisableButtonState(true)
+
+    try {
+      await axios.post("https://email-free-api.herokuapp.com/send-email", formToBeSubmitted)
+      successToastEmitter()
+    } catch (error) {
       if(error) errorToastEmitter()
-    })
+      setDisableButtonState(false)
```

Comment:
The setDisableButtonState needs to be out of the `try/catch` statement.

Reply:
Well, I don't like this way, but ok. I will makes the changes here.